@here/harp-mapview-decoder/lib/ThemedTileDecoder.ts

```diff
@@ -35,9 +38,10 @@ export abstract class ThemedTileDecoder implements ITileDecoder {
         data: ArrayBufferLike,
         tileKey: TileKey,
         projection: Projection
-    ): Promise<DecodedTile> {
+    ): Promise<DecodedTile | undefined> {
         if (this.m_styleSetEvaluator === undefined) {
```

Comment:
Another question, if there is no style available, why do we even request tiles ?

Reply:
Good question...I think it´s based on the one-way communication between DataSource and DataDecoder in a Worker and that a DataSource can be created and added to MapView without a Theme assigned to neither of both, that would contain style information for the Data.
Neither MapView not DataSource does know the state of the decoder and just sends their commands and see what comes back. So if the Decoder did not yet get a theme but gets a request to decode a tile the result of the decoding is undefined.

Reply:
Yes, I see the problem now, there would need to be some two way communication which we don't support. Ok, sounds good.

Reply:
I also wonder if we can avoid sending the payload to the decoder in this case. I mean, we know that the resulting DecodedTile will be empty so why bothering creating the request to decode the tile.